src/algebra/lie/basic.lean

```diff
@@ -1103,12 +1102,23 @@ variables (R : Type u) (L : Type v) (M : Type w)
 variables [comm_ring R] [lie_ring L] [lie_algebra R L] [add_comm_group M] [module R M]
 variables [lie_ring_module L M] [lie_module R L M]
 
-section lie_module
+namespace lie_algebra
+
+/-- The derived series of Lie ideals of a Lie algebra. -/
+def derived_series : ℕ → lie_ideal R L
+| 0       := ⊤
+| (k + 1) := ⁅derived_series k, derived_series k⁆
+
+end lie_algebra
 
-/-- The derived Lie submodule of a Lie module. -/
-def derived_lie_submodule : lie_submodule R L M := ⁅(⊤ : lie_ideal R L), ⊤⁆
```

Comment:
Evidently I am proposing to drop this definition (even though I only just introduced it!). This is because:

1. The name is not really right: there is no derived series for a Lie module. If anything this should be the "lower central submodule".
2. We don't need it since we should have the `derived_series` and `lower_central_series` with these changes.